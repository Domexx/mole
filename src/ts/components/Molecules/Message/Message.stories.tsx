import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import MessageComponent, { Props } from './Message';

import './Message.css';

const Component = {
  title: 'Components/Molecules/Message',
  component: MessageComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof MessageComponent>> = (args) => {
  return <MessageComponent {...args} />;
};

export const Message = Template.bind({});
export default Component;

// https://storybook.js.org/docs/react/essentials/controls
Message.args = {
  text: 'TestText',
  buttonText: 'Play again',
} as Props;
