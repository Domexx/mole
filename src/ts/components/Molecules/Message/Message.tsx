import React, { useContext } from 'react';
import Button from '../../Atoms/Button';
import { GameContext, GameContextData } from '../../../states/GameStateProvider';

export type Props = {
  text: string;
  buttonText: string;
};

const Message: React.FC<Props> = ({ text, buttonText }) => {
  const gameState: GameContextData = useContext(GameContext);

  if (!gameState) {
    return <h1>Game not initialized</h1>;
  }

  return (
    <div className="Message" data-testid="Message">
      <h1>{text}</h1>
      <div className="Message-wrapper">
        <div>Correct: {gameState.state.CorrectPicks}</div>
        <div>Incorrect: {gameState.state.IncorrectPicks}</div>
      </div>
      <Button
        onClick={() => {
          gameState.endGame();
        }}
      >
        {buttonText}
      </Button>
    </div>
  );
};

export default Message;
