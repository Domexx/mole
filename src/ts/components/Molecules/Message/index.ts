import loadable from '@loadable/component';

import './Message.css';

export default loadable(() => {
  return import('./Message');
});
