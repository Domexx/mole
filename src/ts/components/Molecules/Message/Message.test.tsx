import React from 'react';
import { toJson } from '@test';
import renderer from 'react-test-renderer';
import Message from './Message';

test('Matches Snapshot', () => {
  const component = renderer.create(<Message text="TestText" buttonText="TestText" />);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
