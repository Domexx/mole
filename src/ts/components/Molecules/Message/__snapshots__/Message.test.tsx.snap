// Vitest Snapshot v1

exports[`Matches Snapshot 1`] = `
<div
  className="Message"
  data-testid="Message"
>
  <h1>
    TestText
  </h1>
  <div
    className="Message-wrapper"
  >
    <div>
      Correct: 
      0
    </div>
    <div>
      Incorrect: 
      0
    </div>
  </div>
</div>
`;
